param/parameterized.py

```diff
@@ -3744,7 +3755,14 @@ class Foo(Parameterized):
     def __init__(self, **params):
         global object_count
 
-        self._param__private = _InstancePrivate()
+        # Setting a Parameter in an __init__ block before calling super().__init__
```

Comment:
Seems scary, what happens for a admittedly contrived case like this:

```python
class A(param.Parameterized):

    value = param.String(default="A")

    def __init__(self, depth=0):
       self.value = 'B'
        if depth < 2:
            self.sub = A()
        super().__init__()
```

Reply:
Ok indeed that's plain wrong!